models/git/commit_status.go

```diff
@@ -397,36 +397,16 @@ func GetLatestCommitStatusForRepoCommitIDs(ctx context.Context, repoID int64, co
 
 // FindRepoRecentCommitStatusContexts returns repository's recent commit status contexts
 func FindRepoRecentCommitStatusContexts(ctx context.Context, repoID int64, before time.Duration) ([]string, error) {
-	type result struct {
-		Index int64
-		SHA   string
-	}
-	getBase := func() *xorm.Session {
-		return db.GetEngine(ctx).Table(&CommitStatus{}).Where("repo_id = ?", repoID)
-	}
-
 	start := timeutil.TimeStampNow().AddDuration(-before)
-	results := make([]result, 0, 10)
 
-	sess := getBase().And("updated_unix >= ?", start).
-		Select("max( `index` ) as `index`, sha").
-		GroupBy("context_hash, sha").OrderBy("max( `index` ) desc")
-
-	err := sess.Find(&results)
-	if err != nil {
+	var contexts []string
+	if err := db.GetEngine(ctx).Table("commit_status").
```

Comment:
Maybe we need a sort?

Reply:
Which sorting rule should we use? Maybe we can sort them in chronological order?

Reply:
Maybe `updated_unix` desc?

Reply:
I think here we don't need to use `updated_unix desc`, because
- users don't care about the order of contexts, they just need to know which contexts are matched
- it will make the query more complicated than necessary

If we want users to see contexts in a consistent order, maybe we can sort them in lexicographical order。